### PR TITLE
[ClickUp] close #14111 (new command) + utilize `usePromise` + remove use of "any" + add List Sections + migrate

### LINKED ISSUE #14111: [ClickUp - Tasks Explorer] | Search ClickUp Docs
### Extension

https://www.raycast.com/aparandeh/manage-clickup-tasks

### Description

It would be useful to extend the extension to search ClickUp Docs. 

Many users store information in ClickUp Docs, and being able to search these docs directly from Raycast would be awesome.

I would love to be able to search through the list of all ClickUp docs but also search content in individual docs. 

### Who will benefit from this feature?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by xmok**
## Description

The linter kept throwing errors as this extension was using `any` and is using older version of api. I migrated deps and removed use of "any" to fix this. For refactoring, previously `useEffect` and `useState` were being used which I was able to replace with a single new hook `useClickUp` that uses `usePromise` and wraps the Axios object.

CHANGELOG:
### [Update - New Command] - {PR_MERGE_DATE}

- New `Docs Explorer` command - for now it shows docs and their pages (closes #14111)
- Add List Sections to make it more evident where you are
- add metadata images

### Dev Notes
- Refactor by replacing previous hooks with a `usePromise` for unified error handling and loading states
- Migrate ray and related
- Improve types by minimizing use of "any"
- `useForm` in `Capture` command

## Screencast

https://github.com/user-attachments/assets/0e4db7ba-8f33-4d8f-9eba-40b6aa200bcd

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/clickup/src/types/task.dt.ts [hunk 1]
```diff
@@ -22,17 +22,17 @@ export interface Task {
   points: null;
   time_estimate: null;
   time_spent: number;
-  custom_fields: any[];
-  dependencies: any[];
-  linked_tasks: any[];
+  custom_fields?: CustomField;
+  // dependencies: any[];
```
Comment (Ali-Parandeh) [inline extensions/clickup/src/types/task.dt.ts:26]: Commented out code?

File: extensions/clickup/src/views/DocList/ListDocs.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { Action, ActionPanel, Icon, List } from "@raycast/api";
+import useDocs from "../../hooks/useDocs";
+import ListDocPages from "./ListDocPages";
+
+export default function ListDocs({ workspaceId, workspaceName }: { workspaceId: string; workspaceName: string }) {
```
Comment (Ali-Parandeh) [inline extensions/clickup/src/views/DocList/ListDocs.tsx:5]: Does this need to be default exported? Other functions have not been

**Review by Ali-Parandeh — APPROVED** [review]

**Comment by Ali-Parandeh** [comment]
Really great work upgrading the extension and fixing the issues with @xmok Thank you 🙏

File: extensions/clickup/src/views/DocList/ListDocs.tsx (hunk 2 above)
Comment (xmok) [reply extensions/clickup/src/views/DocList/ListDocs.tsx]: Muscle memory 🫠.
Fixed.

File: extensions/clickup/src/types/task.dt.ts (hunk 1 above)
Comment (xmok) [reply extensions/clickup/src/types/task.dt.ts]: I couldn't find proper references for these types in the API docs. But I parsed it from API responses manually right now. Also resolved ✅.

**Review by JoltCode — CHANGES_REQUESTED** [review]
Awesome stuff! Just spotted something though - we should keep the package-lock.json file 😄

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**